tools/editor/scene_tree_dock.cpp

```diff
@@ -1366,6 +1367,25 @@ void SceneTreeDock::_create() {
 
 		editor_data->get_undo_redo().create_action(TTR("Create Node"));
 
+		// Give the node a name with respect to the user's node name casing setting.
+		String new_name;
+		if (edited_scene) {
+			new_name = parent->validate_child_name(child);
+		} else {
+			new_name = child->get_type();
+		}
+		int name_casing = _get_node_name_casing("scenetree_editor/new_node_name_casing");
```

Comment:
That method is a bit overkill. Your enum values are already 0, 1 and 2 respectively, so you just need to do:
```
int name_casing = EditorSettings::get_singleton()->get("scenetree_editor/new_node_name_casing");
```